core/src/main/kotlin/riven/core/service/workflow/engine/coordinator/WorkflowCoordinationService.kt

```
@@ -0,0 +1,397 @@
+
+            logger.info { "Executing batch of ${readyNodes.size} ready nodes" }
+
+            readyNodes.map { node ->
+                logger.info { "Executing node ${node.id} (${node.type})" }
+
+                // Execute node using existing executeNode logic
+                val result = executeNode(node.id, workspaceId)
+
+                // Check if node failed
+                if (result.status == WorkflowStatus.FAILED) {
+                    throw RuntimeException("Node ${node.id} failed: ${result.error}")
```

detekt (1.23.8)

[warning] RuntimeException is a too generic Exception. Prefer throwing specific exceptions that indicate a specific error case.

(detekt.exceptions.TooGenericExceptionThrown)